osu.Game/Rulesets/Mods/ICanBeToggledDuringReplay.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Rulesets.Mods
+{
+    /// <summary>
+    /// An interface for mods that can be toggle during replay.
+    /// </summary>
+    public interface ICanBeToggledDuringReplay : IApplicableMod
+    {
+        /// <summary>
+        /// A property to set whether the mod has been disabled.
+        /// </summary>
+        bool IsDisable
+        {
+            get;
+        }
+
+        /// <summary>
+        /// Method called when mod toggle.
+        /// </summary>
+        void OnToggle();
```

Comment:
You may just want one single `BindableBool` in this interface, which can handle both value and event flow.

Reply:
Sure, but I'm a little confused where the event is registered
There are some mods that don't have a constructor

Define an event registration method in an interface?

Reply:
What "event"? You mean the `BindValueChanged` callback of the proposed bindable in question?

Mods that don't *currently* have a constructor can receive one. I'm not sure what the problem is here.

Reply:
I have decided to use a method to register events and this method will be called once on load